packages/app/src/resource/ToolsPage.test.tsx

```diff
@@ -213,60 +245,347 @@ describe('ToolsPage', () => {
+    // See: https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Portal/Portal.tsx
     await act(async () => {
+      await sleep(150);
+    });
+
+    expect(
+      await screen.findByText(
+        'Unable to determine the current version of the agent. Check the network connectivity of the agent.'
+      )
+    ).toBeInTheDocument();
+
+    expect(screen.queryByRole('button', { name: /confirm upgrade/i })).not.toBeInTheDocument();
+  });
+
+  test('Upgrade -- Error', async () => {
+    clearReleaseCache();
+    globalThis.fetch = mockFetch(200, (url) => {
+      if (url.startsWith(`${GITHUB_RELEASES_URL}/latest`)) {
+        return {
+          tag_name: 'v3.2.14',
+          assets: [
+            {
+              url: 'https://api.github.com/repos/medplum/medplum/releases/assets/193665170',
+              id: 193665170,
+              name: 'medplum-agent-3.2.14-linux',
+              browser_download_url:
+                'https://github.com/medplum/medplum/releases/download/v3.2.14/medplum-agent-3.2.14-linux',
+            },
+          ],
+        };
+      }
+
+      throw new Error('Expected Github releases URL to be called');
+    });
+
+    medplum = new MockClient();
+    medplum.router.router.add('GET', 'Agent/:id/$status', async () => [
+      allOk,
+      {
+        resourceType: 'Parameters',
+        parameter: [
+          {
+            name: 'status',
+            valueCode: 'connected',
+          },
+          {
+            name: 'version',
+            valueString: '3.2.13',
+          },
+          {
+            name: 'lastUpdated',
+            valueCode: new Date().toISOString(),
+          },
+        ],
+      },
+    ]);
+    medplum.router.router.add('GET', 'Agent/:id/$upgrade', async () => {
+      console.log('called');
+      return [serverError(new Error('Something is broken'))];
+    });
+
+    const medplumGetSpy = jest.spyOn(medplum, 'get');
+
+    agent = await medplum.createResource<Agent>({
+      resourceType: 'Agent',
+      name: 'Agente - Upgrade error',
+      status: 'active',
+    });
+
+    setup(`/${getReferenceString(agent)}/tools`);
+
+    expect((await screen.findAllByText(agent.name))[0]).toBeInTheDocument();
+
+    act(() => {
       fireEvent.click(screen.getByRole('button', { name: /upgrade/i }));
     });
 
-    await expect(screen.findByText('Error')).resolves.toBeInTheDocument();
+    // This sleep is load bearing
+    // Basically there is some strange behavior around the Mantine Portal implementation where it is initially rendered as `null`
+    // The theory is that the above `act` is unable to track the Modal children since they are initially not rendered and therefore their useEffects
+    // Are not queued before the end of the `act` block
+    // See: https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Portal/Portal.tsx
+    await act(async () => {
+      await sleep(150);
```

Comment:
Oof.  If it was just one `sleep()`, I might be able to swallow it, but repeating it for every test feels wrong.

Do we need to use Mantine Portal?

Reply:
Did you try using `await act(async () ... `?

Reply:
Not sure how but removing those `act`s works now....

It truly is a mystery because I spent probably 4 hours trying to get that to work on Wednesday

Reply:
Okay, so now the tests pass locally but still fail in CI:
https://github.com/medplum/medplum/actions/runs/11116183748/job/30886228678?pr=5305

Going to try adding back the `await act(async () => {});`

Reply:
Yep not even that works. Load-bearing sleeps seem to be necessary.

I don't fully understand why except I noticed that some of the effects that change the rendering of the modal children are not able to run, no matter how long you sleep in the first `act`, it seems like the updates in the first render have to be flushed before you can continue and trigger the effects in `UpgradeConfirmContent`.